ArduCopter/AP_Arming.cpp

```diff
@@ -1,10 +1,5 @@
 #include "Copter.h"
 
-#pragma GCC diagnostic push
-#if defined(__clang__)
```

Comment:
This warning was added in Clang v14. Better to update the check rather than remove it, e.g.:
```
#if defined(__clang_major__) && __clang_major__ >= 14
```

Reply:
Okay, thanks, I will update the check then instead of removing it.